xarray/conventions.py

```diff
@@ -837,11 +838,12 @@ def decode_cf_variables(variables, attributes, concat_characters=True,
 
     def stackable(dim):
         # figure out if a dimension can be concatenated over
-        if dim in variables:
+        if dim in variables and dim not in concat_dimensions:
```

Comment:
Yes, this logic looks right to me.

Adding it to open_dataset also seems OK, but let's pick a more descriptive name that makes it clear it applies to character/string arrays, e.g., `concat_char_dimensions`?